public/app/features/variables/textbox/TextBoxVariablePicker.tsx

```diff
@@ -1,43 +1,47 @@
-import React, { ChangeEvent, FocusEvent, KeyboardEvent, PureComponent } from 'react';
+import React, { ChangeEvent, FocusEvent, KeyboardEvent, ReactElement, useCallback, useState } from 'react';
 
 import { TextBoxVariableModel } from '../types';
-import { toVariableIdentifier, toVariablePayload } from '../state/types';
-import { dispatch } from '../../../store/store';
+import { toVariablePayload } from '../state/types';
 import { changeVariableProp } from '../state/sharedReducer';
 import { VariablePickerProps } from '../pickers/types';
-import { updateOptions } from '../state/actions';
+import { Input } from '@grafana/ui';
+import { variableAdapters } from '../adapters';
+import { useDispatch } from 'react-redux';
 
 export interface Props extends VariablePickerProps<TextBoxVariableModel> {}
 
-export class TextBoxVariablePicker extends PureComponent<Props> {
-  onQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
-    dispatch(
-      changeVariableProp(toVariablePayload(this.props.variable, { propName: 'query', propValue: event.target.value }))
-    );
-  };
+export function TextBoxVariablePicker({ variable }: Props): ReactElement {
+  const {
+    id,
+    type,
+    current: { value },
+  } = variable;
 
-  onQueryBlur = (event: FocusEvent<HTMLInputElement>) => {
-    if (this.props.variable.current.value !== this.props.variable.query) {
-      dispatch(updateOptions(toVariableIdentifier(this.props.variable)));
-    }
-  };
+  const dispatch = useDispatch();
+  const [updatedValue, setUpdatedValue] = useState(value);
 
-  onQueryKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
-    if (event.keyCode === 13 && this.props.variable.current.value !== this.props.variable.query) {
-      dispatch(updateOptions(toVariableIdentifier(this.props.variable)));
+  const updateVariable = useCallback(() => {
```

Comment:
Same comment about memoization as the one about `onKeyDown` handler :)

Reply:
Fixed it, thank you!